lib/stackprof/report.rb

```diff
@@ -80,65 +83,36 @@ def print_stackcollapse
       end
     end
 
-    def print_flamegraph(f=STDOUT, skip_common=true)
+    def view_flamegraph_in_browser(filepath)
       raise "profile does not include raw samples (add `raw: true` to collecting StackProf.run)" unless raw = data[:raw]
+      profile_b64 = Base64.strict_encode64(JSON.generate(data))
+      filename = File.basename(filepath)
+      js_source = "speedscope.loadFileFromBase64('#{filename}', '#{profile_b64}')"
 
-      stacks = []
-      max_x = 0
-      max_y = 0
-      while len = raw.shift
-        max_y = len if len > max_y
-        stack = raw.slice!(0, len+1)
-        stacks << stack
-        max_x += stack.last
-      end
-
-      f.puts 'flamegraph(['
-      max_y.times do |y|
-        row_prev = nil
-        row_width = 0
-        x = 0
-
-        stacks.each do |stack|
-          weight = stack.last
-          cell = stack[y] unless y == stack.length-1
-
-          if cell.nil?
-            if row_prev
-              flamegraph_row(f, x - row_width, y, row_width, row_prev)
-            end
-
-            row_prev = nil
-            x += weight
-            next
-          end
+      tmpdir = Dir.mktmpdir("stackprof")
 
-          if row_prev.nil?        # start new row with this cell
-            row_width = weight
-            row_prev = cell
-            x += weight
-
-          elsif row_prev == cell  # grow current row along x-axis
-            row_width += weight
-            x += weight
+      tmp_js_path = File.join(tmpdir, "profile.js")
+      File.open(tmp_js_path, "w") do |tmp_js_file|
+        tmp_js_file.write(js_source)
+      end
+      puts "Creating temp file #{tmp_js_path}"
 
-          else                    # end current row and start new row
-            flamegraph_row(f, x - row_width, y, row_width, row_prev)
-            x += weight
-            row_prev = cell
-            row_width = weight
-          end
+      speedscope_path = File.join(File.expand_path(File.dirname(__FILE__)), '..', '..', 'vendor', 'speedscope', 'speedscope', 'index.html')
 
-          row_prev = cell
-        end
-
-        if row_prev
-          next if skip_common && row_width == max_x
+      # For some silly reason, the OS X open command ignores any query parameters or hash parameters
+      # passed as part of the URL. To get around this weird issue, we'll create a local HTML file
+      # that just redirects.
+      tmp_html_path = File.join(tmpdir, "redirect.html")
+      File.open(tmp_html_path, "w") do |tmp_html_file|
+        tmp_html_file.write("<script>window.location='#{speedscope_path}\#localProfilePath=#{tmp_js_path}'</script>'")
+      end
+      puts "Creating temp file #{tmp_html_path}"
 
-          flamegraph_row(f, x - row_width, y, row_width, row_prev)
-        end
+      if not `which xdg-open`.empty?
```

Comment:
To avoid shelling out too much, could you implement this:

https://stackoverflow.com/a/5471032/3574689

Not that `stackprof` has Windows support anyway, but would save this being something needing to be changed in the future if that ever becomes the case.

Reply:
The commands being checked (`open` and `xdg-open`) here are unlikely to be implemented in Windows in the first place, so I'm not sure if there's much benefit. This will still need to be modified for Windows support to run some command that will successfully open the browser, so I'm not sure if the proposed solution helps much on Windows.

Reply:
Made most of my case over in [this comment](https://github.com/tmm1/stackprof/pull/100/files#r205323846), but did want to emphasize that this and the other comment was more of a suggestion, not a requirement or even a "strongly worded request".